infrastructure/SharedInfrastructure/src/SharedInfrastructure/NetworkStack.cs

```diff
@@ -0,0 +1,39 @@
+using Amazon.CDK;
+using Constructs;
+using Amazon.CDK.AWS.EC2;
+
+namespace SharedInfrastructure.Production;
+
+public class NetworkStack : Stack
+{
+    public Vpc Vpc { get; private set; }
+
+    internal NetworkStack(Construct scope, string id, IStackProps props = null) : base(scope, id, props)
+    {
+        // Create a new vpc spanning two AZs and with public and private subnets
+        // to host the application resources
+        Vpc = new Vpc(this, $"{Constants.AppName}Vpc", new VpcProps
+        {
+            IpAddresses = IpAddresses.Cidr("10.0.0.0/16"),
+            // Cap at 2 AZs in case we are deployed to a region with only 2
+            MaxAzs = 2,
+            // We do not need a NAT gateway for this sample application, so remove to reduce cost
```

Comment:
Comment needs to be deleted since we confirmed we need a gateway or vpc endpoints for this to work when hosted in apprunner. Maybe add a comment on why we chose a nat gateway (both approaches cost money).

Reply:
When we pull the AppRunner stack in I will try to configure the NAT gateway within that stack. That way developers aren't incurring NAT Gateway costs if they are not deploying to AppRunner.